### PR TITLE
travis: enable go-{1.5,1.6,tip} and tests

### PR DESCRIPTION
**Pull request by steveeJ**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+
+matrix:
+  allow_failures:
+    - go: tip
 
 install:
  - go get golang.org/x/tools/cmd/cover
 
 script:
  - ./build-check
```
Comment (jonboulle) [inline .travis.yml]: build-check and test are largely redundant. Can we combine them?

Comment (steveeJ) [inline .travis.yml]: AFAICT build-check effectively just runs `gofmt`, so IMHO we can get rid of it all-together.

**Comment by steveeJ** [comment]
@jonboulle all green now, PTAL

File: test [hunk 2]
```diff
@@ -47,4 +39,12 @@ if [ -n "${fmtRes}" ]; then
 	exit 255
 fi
 
+# split TEST into an array and prepend REPO_PATH to each local package
+split=(${TEST// / })
+TEST=${split[@]/#/${REPO_PATH}/}
+
+echo "Running tests..."
+go test -i ${TEST}
+sudo -E bash -c "umask 0; PATH=\$GOROOT/bin:\$PATH go test ${COVER} $@ ${TEST}"
```
Comment (jonboulle) [inline test]: Why do we need umask?

**Comment by jonboulle** [comment]
Should we protect/warn users about running this locally at all?

File: test (hunk 2 above)
Comment (steveeJ) [inline test]: It's for a normal user to be able to clean temporary files. Travis probably doesn't it that as it runs a instance every time, it matters for local runs.

**Comment by steveeJ** [comment]
> Should we protect/warn users about running this locally at all?

It doesn't hurt to add a warning, even though a password prompt should indicate danger for most people and maybe have them look at the test file ;-)

**Comment by jonboulle** [comment]
Yeah but e.g. if someone has sudo cached, or doesn't have a password..

**Comment by steveeJ** [comment]
> Yeah but e.g. if someone has sudo cached, or doesn't have a password..

Yep, it's safe now!

**Comment by jonboulle** [comment]
Is there any more "standard" way we can do this than SUDO_PERMITTED?

**Comment by steveeJ** [comment]
> Is there any more "standard" way we can do this than SUDO_PERMITTED?

I thought of using the `CI` env var, but I deicded for something more explicit. Suggestions are welcome.

**Comment by jonboulle** [comment]
I'm not convinced that the test script should _fail_ without it, rather than just warn.

**Comment by jonboulle** [comment]
I guess CNI has the same problem: https://github.com/appc/cni/blob/master/test#L38

**Comment by steveeJ** [comment]
What good is a warning just before you get shot?

**Comment by jonboulle** [comment]
Huh? I don't mean before running the sudo command. I mean that the script
shouldn't return 1 if not invoked as sudo, it should just test whatever it
can and warn that sudo is required for all tests to run. But this feels
like it's turning into a bike shed.

**Comment by jonboulle** [comment]
The colour is green.

**Pull request merged**